### DISCUSSION TITLE
Return value not checked - defer zapLog.Sync()

### DISCUSSION THREAD

**Discussion by motawy-citrus** [uber-go/zap#1203] [Q&A]
I recently dived into the world of Go, so apologies in advance if it's something really easy.

I'm receiving this error from the `golangci-lint` 

```
Error return value of `zapLog.Sync` is not checked (errcheck)
	defer zapLog.Sync()
```

Here's what I do to define the logger: 
```go
var zapConfig zap.Config
zapConfig = zap.NewProductionConfig()
zapConfig.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
zapLog, err = zapConfig.Build(zap.AddCallerSkip(1))
if err != nil {
	panic(err)
}
defer zapLog.Sync()
```

Is there a best practice or a simple way to error check in this case? 

I've tried with a simple closure but I get a different issue (container fails but no clear message on the error)

**Comment by prashantv** [comment]
You should be able to get past this error by assigning the error to `_`,
```go
defer func() {
  _ = zaplog.Sync()
}()
```

However, it's probably less noisy to use `//nolint:errcheck` on that line,
```go
defer zaplog.Sync() //nolint:errcheck
```
